engine/api/workflow/workflow_run_event.go

```diff
@@ -24,91 +24,111 @@ func ResyncCommitStatus(ctx context.Context, db gorp.SqlExecutor, store cache.St
 	)
 	defer end()
 
-	for nodeID, nodeRuns := range wr.WorkflowNodeRuns {
+	commitsMap := make(map[string][]sdk.VCSCommitStatus)
+	for _, nodeRuns := range wr.WorkflowNodeRuns {
 		sort.Slice(nodeRuns, func(i, j int) bool {
 			return nodeRuns[i].SubNumber >= nodeRuns[j].SubNumber
 		})
 		nodeRun := nodeRuns[0]
-
-		if !sdk.StatusIsTerminated(nodeRun.Status) {
-			continue
+		var err error
+		commitsMap, err = SendVCSEvent(ctx, db, store, proj, *wr, nodeRun, commitsMap)
```

Comment:
I understand the commitsMap thing, but i think it could be nice to instanciate an object (`VCSEventMessenger` ?) that holds and hides this `commitsMap` + the vcsClient and then put the `SendVCSEvent` function as a method of this object